File: api/vehicle-service/src/main/java/com/carrental/vehicleservice/model/dto/AvailableVehiclesSearchDTO.java
package com.carrental.vehicleservice.model.dto;

import lombok.Data;

import java.io.Serializable;
import java.util.Date;

@Data
public class AvailableVehiclesSearchDTO implements Serializable {

    private Long locationId;

    private Date receiptDate;

    private Date returnDate;
}

File: frontend-app/shared/model/AvailableVehiclesSearchDTO.ts
export class AvailableVehiclesSearchDTO {
    locationId: string;
    receiptDate: string;
    returnDate: string;

    constructor(locationId: string, receiptDate: string, returnDate: string) {
        this.locationId = locationId;
        this.receiptDate = receiptDate;
        this.returnDate = returnDate;
    }
}
